src/main/webapp/app/modules/canvas/visualization/util/visualization-render-utils.ts

```diff
@@ -41,232 +41,330 @@ export const buildQueryDTO = (visualMetaData, filter) => {
         .data(metaData);
       infographicChartObj(div[0]);
     },
   },
 };
 
-export const renderVisualization = (visual, metaData, element = 'widget') => {
-  if (visual.fields && ValidateFields(visual.fields)) {
-    const widget = $(`#${element}-${visual.id}`);
-    if (widget.length > 0) {
-      let height = widget[0].clientHeight;
-      const width = widget[0].clientWidth;
-      const chartId = `chart-${element}-${visual.id}`;
+export const renderIframe = (item, height, widget) => {
+  const iframeLink = item.properties[0].value;
 
-      $('#' + chartId).remove('');
-      if (element === 'widget') {
-        height = height - 30;
-        $('#visualization-' + visual.id).append(
-          '<div id="' +
-            chartId +
-            '" height="' +
-            height +
-            '" width="' +
-            width +
-            '" style="width:' +
-            width +
-            'px; height:' +
-            height +
-            'px;overflow:hidden;position:relative" ></div>'
-        );
-      } else {
-        $('#visualization-edit-' + visual.id).append(
-          '<div id="' +
-            chartId +
-            '" height="' +
-            height +
-            '" width="' +
-            width +
-            '" style="width:' +
-            width +
-            'px; height:' +
-            height +
-            'px;overflow:hidden;position:relative" ></div>'
-        );
-      }
+  $(`#iframe-${item.id}`)
+    .attr('width', widget - 30)
+    .attr('height', height - 30)
+    .attr('src', iframeLink);
+};
 
-      const div = $('#' + chartId);
+export const renderVisualization = (visual, metaData, element = 'widget', props = null) => {
+  const widget = $(`#${element}-${visual.id}`);
+  let height = widget[0].clientHeight;
+  const width = widget[0].clientWidth;
+  if (widget.length > 0) {
+    if (visual.metadataVisual.name === VisualizationType.Iframe) {
+      renderIframe(visual, height, width);
+    } else {
+      if (visual.fields && ValidateFields(visual.fields)) {
+        let height = widget[0].clientHeight;
+        const width = widget[0].clientWidth;
+        const chartId = `chart-${element}-${visual.id}`;
 
-      drawVisualization[visual.metadataVisual.name].drawChart(visual, metaData, div);
+        $('#' + chartId).remove('');
+        if (element === 'widget') {
+          height = height - 30;
+          $('#visualization-' + visual.id).append(
+            '<div id="' +
+              chartId +
+              '" height="' +
+              height +
+              '" width="' +
+              width +
+              '" style="width:' +
+              width +
+              'px; height:' +
+              height +
+              'px;overflow:hidden;position:relative" ></div>'
+          );
+        } else {
+          $('#visualization-edit-' + visual.id).append(
+            '<div id="' +
+              chartId +
+              '" height="' +
+              height +
+              '" width="' +
```

Comment:
why using jquery

Reply:
its only use for fetch element as we already discussed earlier